File: src/fasta_parser.pl
#!/usr/bin/env perl

=pod

=head2

Read fasta formatted input from stdin and print tab separated id and sequence to stdout using pure perl (no bioperl)

=cut

use strict;
use warnings;

my $term = $/;
my %seq_hash = ();
$/ = ">";
while (<>) {
    chomp;
    next if($_ eq '');
    my ($id, @sequencelines) = split /\n/;
    foreach my $line (@sequencelines) {
        $seq_hash{$id} .= $line;
    }
}
$/ = $term;
while ( my ($header, $sequence) = each %seq_hash ) {
    print $header, "\t", $sequence, "\n";
}

File: src/fasta_parser.py
#! /usr/bin/env python3

"""
Read fasta formatted input from stdin and print tab separated id and sequence to stdout using pure python (no biopython)
"""

import sys

def main():
    seq = ''
    for line in sys.stdin:
        if line[0] == '>':
            if seq:
                print(label + '\t' + seq)
            label = line[1:].rstrip()
            seq = ''
        else:
            seq += line.rstrip()
    print(label + '\t' + seq)

if __name__ == "__main__":
    main()

